hydrolib/core/io/polyfile/serializer.py

```diff
@@ -0,0 +1,79 @@
+from pathlib import Path
+from typing import Dict
+from hydrolib.core.io.polyfile.components import (
+    Description,
+    Metadata,
+    Point,
+    PolyObject,
+)
+
+
+class Serializer:
+    @staticmethod
+    def serialize_description(description: Description) -> str:
+        """Serialize the Description to a string which can be used within a polyfile.
+
+        Returns:
+            str: The serialised equivalent of this Description
+        """
+        if description.content == "":
+            return "*"
+        else:
+            return "\n".join(f"*{v.rstrip()}" for v in description.content.splitlines())
+
+    @staticmethod
+    def serialize_metadata(metadata: Metadata) -> str:
+        """Serialize this Metadata to a string which can be used within a polyfile.
+
+        The number of rows and number of columns are separated by four spaces.
+
+        Returns:
+            str: The serialised equivalent of this Metadata
+        """
+        return f"{metadata.name}\n{metadata.n_rows}    {metadata.n_columns}"
+
+    @staticmethod
+    def serialize_point(point: Point) -> str:
+        """Serialize this Point to a string which can be used within a polyfile.
+
+        the point data is indented with 4 spaces, and the individual values are
+        separated by 4 spaces as well.
+
+        Returns:
+            str: The serialised equivalent of this Point
+        """
+        z_val = f"{point.z}    " if point.z is not None else ""
+        data_vals = "    ".join(str(v) for v in point.data)
+        return f"    {point.x}    {point.y}    {z_val}{data_vals}".rstrip()
```

Comment:
Shouldn't there be a space in there? Or four spaces?
```suggestion
        return f"    {point.x}    {point.y}    {z_val}    {data_vals}".rstrip()
```

Reply:
this would case a problem in case we do not have a `point.z` but do have `data_vals` then we would have 8 whitespaces, which is not consistent, thus I put the four spaces in the `z_val` serializing if it is not none, and else an empty string. I do agree that it is not directly obvious, but I do not see a better way.